### ISSUE TITLE
pre-annotations not showing in multi page document annotation

### ISSUE THREAD

**Issue by adsk2050** [heartexlabs/label-studio#5025]
The pre-annotations I am uploading are not showing in the labeling interface.

Here is the context about my project:
- I am doing Multi Page OCR. 
- Label studio is hosted on a docker container.
- One task in one json file. Imported by uploading in frontend. No errors at import.
- Document images are stored on S3, S3 is integrated with label studio and working properly.
- Task is getting created and listed, images are loading, but there are no annotations in the "Regions" tab.

Label config:
```
<View>
    <Repeater on="$document" indexFlag="{{idx}}" mode="pagination">
        <Image name="page_{{idx}}" value="$document[{{idx}}].page_img_link" valueType="url"/>
        <Labels name="label_{{idx}}" toName="page_{{idx}}">
            <Label value="random_label" background="#69a92f"/>
            <Label value="another_random_label" background="#e31d69"/>
        </Labels>
        <Rectangle name="bbox_{{idx}}" toName="page_{{idx}}" strokeWidth="3"/>
        <TextArea name="transcription_{{idx}}" toName="page_{{idx}}" editable="true" perRegion="true" required="true" maxSubmissions="10" rows="25" placeholder="Recognized Text" displayMode="region-list"/>
    </Repeater>
</View>
```
Sample task:
```
[
  {
    "annotations": [
      {
        "result": [
          {
            "original_width": 1654,
            "original_height": 2340,
            "image_rotation": 0,
            "value": {
              "x": 48.851269649334945,
              "y": 53.034188034188034,
              "width": 2.4183796856106445,
              "height": 0.7692307692307665,
              "rotation": 0
            },
            "id": "9ca0409a0a",
            "from_name": "bbox_4",
            "to_name": "image_4",
            "type": "rectangle",
            "origin": "manual"
          },
          {
            "original_width": 1654,
            "original_height": 2340,
            "image_rotation": 0,
            "value": {
              "x": 48.851269649334945,
              "y": 53.034188034188034,
              "width": 2.4183796856106445,
              "height": 0.7692307692307665,
              "rotation": 0,
              "labels": [
                "random_label"
              ]
            },
            "id": "9ca0409a0a",
            "from_name": "label_4",
            "to_name": "image_4",
            "type": "labels",
            "origin": "manual"
          },
          {
            "original_width": 1654,
            "original_height": 2340,
            "image_rotation": 0,
            "value": {
              "x": 48.851269649334945,
              "y": 53.034188034188034,
              "width": 2.4183796856106445,
              "height": 0.7692307692307665,
              "rotation": 0,
              "text": [
                "Random Label text"
              ]
            },
            "id": "9ca0409a0a",
            "from_name": "transcription_4",
            "to_name": "image_4",
            "type": "textarea",
            "origin": "manual"
          }
        ]
      }
    ],
    "predictions": [],
    "data": {
      "document": [
        {
          "page_img_link": "s3://custom_s3_bucket/page_00.jpg"
        },
        {
          "page_img_link": "s3://custom_s3_bucket/page_01.jpg"
        },
        {
          "page_img_link": "s3://custom_s3_bucket/page_02.jpg"
        },
        {
          "page_img_link": "s3://custom_s3_bucket/page_03.jpg"
        },
        {
          "page_img_link": "s3://custom_s3_bucket/page_04.jpg"
        },
        {
          "page_img_link": "s3://custom_s3_bucket/page_05.jpg"
        }
      ],
      "poc": "anmol",
      "relevant_pages": [
        "4"
      ]
    }
  }
]
```

**Expected behavior**
The pre-annotations should get listed in their respective pages.

**Environment:**
 - OS: ubuntu
 - Label Studio Version: 1.8.2

**Additional context**

**Comment by adsk2050** [comment]
This is what comes as a response of the API call to `/api/tasks/423?project=31` looks similar to (redacted sensitive info):
```
{
    "id": 423,
    "predictions": [],
    "annotations": [
        {
            "id": 4222,
            "created_username": ", 1",
            "created_ago": "1 day, 2 hours",
            "completed_by": {
                "id": 1,
                "first_name": "",
                "last_name": "",
                "avatar": null,
                "email": "",
                "initials": "an"
            },
            "result": [
                {
                    "original_width": 1654,
                    "original_height": 2340,
                    "image_rotation": 0,
                    "value": {
                        "x": 48.851269649334945,
                        "y": 53.034188034188034,
                        "width": 2.4183796856106445,
                        "height": 0.7692307692307665,
                        "rotation": 0
                    },
                    "id": "543e4358-2",
                    "from_name": "bbox_4",
                    "to_name": "image_4",
                    "type": "rectangle",
                    "origin": "manual"
                },
                {
                    "original_width": 1654,
                    "original_height": 2340,
                    "image_rotation": 0,
                    "value": {
                        "x": 48.851269649334945,
                        "y": 53.034188034188034,
                        "width": 2.4183796856106445,
                        "height": 0.7692307692307665,
                        "rotation": 0,
                        "labels": [
                            "random_label"
                        ]
                    },
                    "id": "543e4358-2",
                    "from_name": "label_4",
                    "to_name": "image_4",
                    "type": "labels",
                    "origin": "manual"
                },
                {
                    "original_width": 1654,
                    "original_height": 2340,
                    "image_rotation": 0,
                    "value": {
                        "x": 48.851269649334945,
                        "y": 53.034188034188034,
                        "width": 2.4183796856106445,
                        "height": 0.7692307692307665,
                        "rotation": 0,
                        "text": [
                            "random text"
                        ]
                    },
                    "id": "543e4358-2",
                    "from_name": "transcription_4",
                    "to_name": "image_4",
                    "type": "textarea",
                    "origin": "manual"
                },
                {
                    "original_width": 1654,
                    "original_height": 2340,
                    "image_rotation": 0,
                    "value": {
                        "x": 51.692865779927445,
                        "y": 53.034188034188034,
                        "width": 10.580411124546563,
                        "height": 0.7692307692307665,
                        "rotation": 0
                    },
                    "id": "7adf0bb8-6",
                    "from_name": "bbox_4",
                    "to_name": "image_4",
                    "type": "rectangle",
                    "origin": "manual"
                },
                {
                    "original_width": 1654,
                    "original_height": 2340,
                    "image_rotation": 0,
                    "value": {
                        "x": 51.692865779927445,
                        "y": 53.034188034188034,
                        "width": 10.580411124546563,
                        "height": 0.7692307692307665,
                        "rotation": 0,
                        "labels": [
                            "random_label"
                        ]
                    },
                    "id": "7adf0bb8-6",
                    "from_name": "label_4",
                    "to_name": "image_4",
                    "type": "labels",
                    "origin": "manual"
                },
                {
                    "original_width": 1654,
                    "original_height": 2340,
                    "image_rotation": 0,
                    "value": {
                        "x": 51.692865779927445,
                        "y": 53.034188034188034,
                        "width": 10.580411124546563,
                        "height": 0.7692307692307665,
                        "rotation": 0,
                        "text": [
                            "random text"
                        ]
                    },
                    "id": "7adf0bb8-6",
                    "from_name": "transcription_4",
                    "to_name": "image_4",
                    "type": "textarea",
                    "origin": "manual"
                }
            ],
            "was_cancelled": false,
            "ground_truth": true,
            "created_at": "2023-11-08T07:15:11.640385Z",
            "updated_at": "2023-11-08T07:15:11.640415Z",
            "draft_created_at": null,
            "lead_time": null,
            "import_id": null,
            "last_action": null,
            "task": 423,
            "project": 31,
            "updated_by": null,
            "parent_prediction": null,
            "parent_annotation": null,
            "last_created_by": null
        }
    ],
    "drafts": [],
    "annotators": [
        1
    ],
    "inner_id": 1,
    "cancelled_annotations": 0,
    "total_annotations": 1,
    "total_predictions": 0,
    "completed_at": "2023-11-08T07:15:11.640385Z",
    "annotations_results": "<REDACTED>",
    "predictions_results": "",
    "predictions_score": null,
    "file_upload": "729b028a-INDIA_F_11__1646659857683.json",
    "storage_filename": null,
    "annotations_ids": "4222",
    "predictions_model_versions": "",
    "avg_lead_time": null,
    "draft_exists": false,
    "updated_by": [],
    "data": {
        "document": [
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            },
            {
                "page_img_link": "/tasks/423/presign/?fileuri=<some_long_string>"
            }
        ],
        "poc": "anmol",
        "relevant_pages": [
            "5",
            "15",
            "3",
            "6",
            "18"
        ]
    },
    "meta": {},
    "created_at": "2023-11-08T07:15:11.636307Z",
    "updated_at": "2023-11-08T07:15:11.636339Z",
    "is_labeled": true,
    "overlap": 1,
    "comment_count": 0,
    "unresolved_comment_count": 0,
    "last_comment_updated_at": null,
    "project": 31,
    "comment_authors": []
}
```

Clearly, the annotations I am importing are getting saved into the database and sent to the front end. But the **front end is not able to display it**.

The imported annotations are also showing in "Show Task Source" tooltip in the main menu.

---

There is one more interesting piece of information I would like to mention.

I did some annotations manually and exported them. I then copy-pasted the bounding box annotations in the `results` list. And to my surprise, those annotations were getting listed. And from the same list, others, which I had generated programmatically, were not getting listed - despite having the same structure.

**Comment by adsk2050** [comment]
@bmartel @makseq Please help

**Comment by AymanTalkhani** [comment]
@adsk2050Not sure if you need images labeled for segmentation/bounding boxes as well, but if you do, have you tried using [InstaLabel](https://www.instalabel.ai/). All you need to do is upload your dataset, select and label a few samples you'd like to finetune the model on in our platform, and hit the 'Train annotator' button. You can then head over to the 'Annotate' tab, click 'Instant Annotation' and have the finetuned model label the rest of your dataset for you. There's even a crowdsourcing option to correct AI predictions for verified ground truth labels. Could be useful for a lot of custom computer vision labeling use cases if you're on the lookout.

**Comment by hogepodge** [comment]
It looks like you might have misformatted your JSON file for preannotations. Rather than `annotations`, I think you should be using `predictions` as described in [this document.](https://labelstud.io/guide/predictions.html)

Please check that, and if it's not the case we can investigate further.

**Comment by adsk2050** [comment]
Hello @hogepodge 

I used `predictions` instead of `annotations` after your suggestion, but even those are not showing.

In a single image task, I am having no difficulty in using `annotations` as well as `predictions`. I feel it is related to multipage.

**Comment by adsk2050** [comment]
I am noticing that there are dashes in my `id` variable in each of the annotation/prediction, but I never found a single case in which there was as `-` in the `id` of the annotations generated by label studio when I manually annotated. Those contain an underscore `_`.

Could this be the reason my annotations are not working? I will check and update -> **It didnt work**

But I think there is something with the `id` value only as it is the only thing whose value is the odd one out

**Comment by adsk2050** [comment]
Upon going through the console, I got these errors: 
---
`
Error: [mobx-state-tree] You are trying to read or write to an object that is no longer part of a state tree. (Object type: 'Annotation', Path upon death: '/annotationStore/annotations/0', Subpath: 'pk', Action: '/annotationStore.clearDeletedParents()'). Either detach nodes first, or don't use objects after removing / replacing them in the tree.
    at De (mobx-state-tree.module.js:3427:18)
    at t.assertAlive (mobx-state-tree.module.js:1510:21)
    at t.unbox (mobx-state-tree.module.js:1580:14)
    at t.dehanceValue (mobx.module.js:1040:25)
    at t.get (mobx.module.js:1092:21)
    at e.read (mobx.module.js:4136:37)
    at Object.get [as pk] (mobx.module.js:4377:36)
    at Object.clearDeletedParents (store.js:150:24)
    at ve (mobx.module.js:928:19)
    at r (mobx.module.js:915:16)
    at mobx-state-tree.module.js:2593:35
    at q (mobx-state-tree.module.js:2456:16)
    at Object.r (mobx-state-tree.module.js:2483:16)
    at Object.deleteAnnotation (store.js:169:12)
    at ve (mobx.module.js:928:19)
    at r (mobx.module.js:915:16)
    at mobx-state-tree.module.js:2593:35
    at q (mobx-state-tree.module.js:2456:16)
    at Object.r (mobx-state-tree.module.js:2483:16)
    at onOk (AnnotationButton.tsx:104:23)
    at onClick (Modal.js:62:13)
    at Object.Ke (react-dom.production.min.js:52:317)
    at Be (react-dom.production.min.js:52:471)
    at react-dom.production.min.js:53:35
    at Or (react-dom.production.min.js:100:68)
    at jr (react-dom.production.min.js:101:380)
    at react-dom.production.min.js:113:65
    at Ze (react-dom.production.min.js:292:189)
    at react-dom.production.min.js:50:57
    at Lr (react-dom.production.min.js:105:469)
    at qt (react-dom.production.min.js:75:265)
    at Qt (react-dom.production.min.js:74:124)
    at t.unstable_runWithPriority (scheduler.production.min.js:18:343)
    at Ho (react-dom.production.min.js:122:325)
    at Te (react-dom.production.min.js:292:48)
    at Ut (react-dom.production.min.js:73:352)
    at HTMLBodyElement.r (helpers.js:72:23)
(anonymous) @ instrument.js:109
`
---
`instrument.js:109 Error: [mobx-state-tree] You are trying to read or write to an object that is no longer part of a state tree. (Object type: 'Annotation', Path upon death: '/annotations/1', Subpath: 'areas', Action: '/annotations/1.startAutosave()'). Either detach nodes first, or don't use objects after removing / replacing them in the tree.
    at De (mobx-state-tree.module.js:3427:18)
    at t.assertAlive (mobx-state-tree.module.js:1510:21)
    at t.unbox (mobx-state-tree.module.js:1580:14)
    at t.dehanceValue (mobx.module.js:1040:25)
    at t.get (mobx.module.js:1092:21)
    at e.read (mobx.module.js:4136:37)
    at Object.get [as areas] (mobx.module.js:4377:36)
    at Annotation.js:665:23
    at Generator.next (<anonymous>)
    at mobx-state-tree.module.js:3631:51
    at ve (mobx.module.js:928:19)
    at r (mobx.module.js:915:16)
    at mobx-state-tree.module.js:2593:35
    at q (mobx-state-tree.module.js:2456:16)
    at s (mobx-state-tree.module.js:3617:13)
    at c (mobx-state-tree.module.js:3631:21)`

**Comment by adsk2050** [comment]
I have pinpointed the error I was making. I was passing incorrect value in ` "to_name"`. Instead of ` "page_4"`, I used `"image_4"`.
